src/lib/image/image.shim.js

```diff
@@ -37,7 +89,13 @@ var ImageUtil = {
         metadata.image = dimensions;
 
         var ctx = canvas.getContext('2d');
-        ctx.drawImage(img, 0, 0, dimensions.width, dimensions.height);
+        if (file && file.image && file.image.orientation && file.image.orientation !== 1) {
```

Comment:
It is the practice across all clients. We are only altering the thumbnail that gets uploaded to the file share and not the actual original image. It is required because all the clients expect that the thumbnail is correctly rotated before uploaded and so will be displayed properly everywhere.